src/extensions/default/AutoUpdate/main.js

```diff
@@ -90,6 +92,35 @@ define(function (require, exports, module) {
         NETWORK_SLOW_OR_DISCONNECTED: 5
     };
 
+    var updateProgressKey = "autoUpdateInProgress";
+
+    /**
+     * Checks if an auto update session is currently in progress
+     * @returns {boolean} - true if an auto update session is currently in progress, false otherwise
+     */
+    function checkIfAnotherSessionInProgress() {
+        var result = $.Deferred();
+        if(updateJsonHandler) {
+            var state = updateJsonHandler.state;
+
+            updateJsonHandler.read(updateProgressKey)
+                .done(function(val) {
+                    updateJsonHandler.state = state;
+                    setUpdateStateInJSON(updateProgressKey, val)
```

Comment:
why are we setting the value we just read ?

Reply:
Corrected.